jupyter-org-extensions.el

```diff
@@ -0,0 +1,404 @@
+
+If point is in a block, copy the header to the new block"
+  (interactive "P")
+  (if (org-in-src-block-p)
+      (let* ((src (org-element-context))
+             (start (org-element-property :begin src))
+             (end (org-element-property :end src))
+             (lang (org-element-property :language src))
+             (switches (org-element-property :switches src))
+             (parameters (org-element-property :parameters src))
+             location)
+        (if below
+            (progn
+              (goto-char start)
+              (setq location (org-babel-where-is-src-block-result))
+              (if (not location)
+                  (goto-char end)
+                (goto-char location)
+                (goto-char (org-element-property :end (org-element-context))))
+              (insert
+               (org-element-interpret-data
+                (org-element-put-property
+                 (jupyter-org-src-block lang parameters "\n" switches)
+                 :post-blank 1)))
+              (forward-line -3))
+          ;; after current block
+          (goto-char (org-element-property :begin src))
+          (insert
+           (org-element-interpret-data
+            (org-element-put-property
+             (jupyter-org-src-block lang parameters "\n" switches)
+             :post-blank 1)))
+          (forward-line -3)))
+
+    ;; not in a src block, insert a new block, query for jupyter kernel
+    (beginning-of-line)
+    (let ((kernel
+           (completing-read "Kernel: "
+                            (mapcar #'car (jupyter-available-kernelspecs)))))
+      (insert
+       (org-element-interpret-data
+        (org-element-put-property
+         (jupyter-org-src-block
+             (format "jupyter-%s"
+                     (plist-get (cdr (alist-get kernel
+                                                (jupyter-available-kernelspecs)
+                                                nil nil 'equal))
+                                :language)) "" "\n")
+         :post-blank 0))))
+    (forward-line -2)))
+
+;;;###autoload
+(defun jupyter-org-split-src-block (&optional below)
+  "Split the current src block with point in upper block.
+
+With a prefix BELOW move point to lower block."
+  (interactive "P")
+  (let ((src-block (org-element-context)))
+    (unless (eq (org-element-type src-block) 'src-block)
+      (error "Not in a source block"))
+    (beginning-of-line)
+    (org-babel-demarcate-block)
+    (if below
+        (progn
+          (org-babel-next-src-block)
+          (forward-line)
+          (open-line 1))
+      (forward-line -2)
+      (end-of-line))))
+
+;;;###autoload
+(defun jupyter-org-execute-and-next-block (&optional new)
+  "Execute this block and jump or add a new one.
+
+If a new block is created, use the same language, switches and parameters.
+With prefix arg NEW, always insert new cell."
+  (interactive "P")
+  (unless (eq (org-element-type (org-element-context)) 'src-block)
```

Comment:
I think it would be better to use `org-in-src-block-p` here.

Reply:
Of course...